vivisect/qt/symboliks.py

```diff
@@ -260,6 +260,13 @@ def rendSymbolikPath(self, *args, **kwargs):
                     self.memcanvas.addText('\n')
                 continue
 
+            if effect.efftype == EFFTYPE_DEBUG:
```

Comment:
Kinda feels like we might want an additional check box for these (or just keep it showing only under `alleff`. Because `DebugEffect` doesn't actually affect emulator state, so it'd be more confusing to see these here. I'd prefer to keep the split of what shows by default in symboliks to be "Here are the things we know to have big effects on the symbolik state", and then let the all effects checkbox show everything else. It keeps things tidy and focused in the default view.

Side note: we should just carve the `effect.walkTree.....` block into a subfunction. The copypasta here is silly.

Reply:
`DebugEffect` doesn't affect emulator state, but the instruction that *isn't* defined likely would have.... thus, the existence of `DebugEffect` should not be hidden from the user without risking their loss of trust.  i've had this happen very recently.  only because i worked through the details and something didn't look right did I think to enable alleffs.

Reply:
My problem is that is overloads It is a communicator of symbolik state, but there's several we don't display. I still err on the side of not displaying too much, and until the full set of instructions is covered, we're going to have these, and for the most part, they really just clutter the space up. Displaying too much data is just as bad as displaying not enough, because it will cause eyes to glaze over. So I'm not convinced we're losing any kind of trust here when you can just click a box to see them. At worst we can add another to show the debug effects. But I'm against shoving a mountain of text in front of the user when the option to show it is right there.

Reply:
i don't run into too many.  but when i do, i have to know,  there's not "I don't want to know that may symbolik state is completely fucked."  ever.

i'm with you on overload.  the magic is getting just the right information displayed.

i cannot be convinced at this point (having used Symboliks for 10+ years) that these are not important, ever.  if *nothing* else, we need to tag the top of the symbolik view with "some instructions were not supported so this state is erroneous.  check "all effects" to see the details."

but this is easier and better in my opinion.

Reply:
question: how many instructions do you expect to be not supported?  
how many should stay that way?

these are *gaps* that (1) shouldn't be there, and (2) typically dramatically affect the accuracy of SymboliksView.
displaying them adds discrete lines.  they don't show up in the crazy symbolik state lines.  if anything, they'll be *easy* on the eyes.

oh, and we're more likely to implement/fix them if they're a little annoying to everyone.